### PR TITLE
Ensure CookieName is configurable

### PR DESCRIPTION
**Pull request by nquandt**
By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Add the ability to specify the cookie scheme used

### Testing

Create aspnet project with custom cookie.

```csharp
services
  .AddAuthentication()
  .AddCookie("MyCookieScheme", () => {})
  .AddAuth0WebAppAuthentication(options => {
    options.CookieName = "MyCookieScheme";
  });
- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: src/Auth0.AspNetCore.Authentication/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -24,17 +24,40 @@ public static Auth0WebAppAuthenticationBuilder AddAuth0WebAppAuthentication(this
...
+                    {
+                        var authOptions = new Auth0WebAppOptions();
+
+                        configureOptions(authOptions);
+
+                        options.DefaultAuthenticateScheme = authOptions.CookieName;
+                        options.DefaultSignInScheme = authOptions.CookieName;
+                        options.DefaultChallengeScheme = authOptions.CookieName;
+                    });
+            }
```
Comment (frederikprijck) [inline src/Auth0.AspNetCore.Authentication/ServiceCollectionExtensions.cs:58]: I dont think we need this change. If you do not want to our SDK to call `AddAuthentication`, you should not use our ServiceCollectionExtensions, you should use our `AuthenticationBuilderExtensions`. 

Instead, we should focus on just the cookieName in this code. I believe we should be dropping the `setDefaultSchemes` and rollback most of these changes but only ensure we set the cookie to `auth0Options.CookieName`.

File: src/Auth0.AspNetCore.Authentication/Auth0WebAppOptions.cs [hunk 2]
```diff
@@ -11,6 +12,14 @@ namespace Auth0.AspNetCore.Authentication
     /// </summary>
     public class Auth0WebAppOptions
     {
+        /// <summary>
+        /// The name of the cookie scheme to use
+        /// </summary>
+        /// <remarks>
+        /// The default is <see cref="CookieAuthenticationDefaults.AuthenticationScheme"/>
+        /// </remarks>
+        public string CookieName { get; set; } = CookieAuthenticationDefaults.AuthenticationScheme;
```
Comment (frederikprijck) [inline src/Auth0.AspNetCore.Authentication/Auth0WebAppOptions.cs:21]: Would it make sense to name this `CookieAuthenticationScheme` ?

**Review by frederikprijck — CHANGES_REQUESTED** [review]
Thanks for the changes! Added a couple of small comments to ensure we keep the PR focussed.

**Comment by nquandt** [comment]
I made some updates based on your comments! Thanks for reviewing, and thanks for the tip about using the `AuthenticationBuilderExtensions` instead of the `ServiceCollectionExtensions`

**Review by frederikprijck — APPROVED** [review]
Thanks for the work here, I patched the PR to add something that was missing, and added a couple of tests to verify it works as intended.

**Pull request merged**